### ISSUE TITLE
Updating version numbers within ansible playbooks

### ISSUE THREAD

**Issue by dragetd** [renovatebot/renovate#8865]
**What would you like Renovate to be able to do?**

I deploy my infrastructure using Ansible. My playbooks contain the different software versions we use.

When using Containerfiles or docker-compose.yml files, it is already possible to use renovate which which suggest updates of those files. But this does not work for versions specified within the Ansible playbooks.

Take the following example:
```
- name: Download setup script file from upstream
  get_url:
    url: "https://raw.githubusercontent.com/docker-mailserver/docker-mailserver/v8.0.0/setup.sh"
    dest: "/usr/local/bin/setup-mail"
    mode: "ugo=rx"
```

or
```
- name: Provide borgmatic using pip3
  pip:
    name: "borgmatic==1.5.12"
    executable: "pip3"
```

I know this is a meta-level which might not be trivial to implement… not Ansible dependencies themself, but dependenceis introduced by modules of Ansible.

**Did you already have any implementation ideas?**

The managers would be probably something like 'ansible-pip' and 'ansible-url-github' and could re-use the existing python/github-releases datasources and ansible playbook parsing. But of course get_url with a GitHub URL or pip are not the only combinations that one could consider. To not create too many managers, this might be combined into a 'ansible-module' kind of manager. Less flexible to configure, but once enabled, it would work with any new ansible module implemented.

## Alternative:

An alternative which would be a huge change for renovate, but would be extremely flexible could be a kind of meta-annotation in comments. For YAML it would be something like `# renovate-hint: { package-regex: "^.*.com/(.*)/v\d.\d.\d/.*$", version-regex: "^.*docker-mailserver/v(\d\.\d.\d).*$", datasource: github-releases }` which I would add before the URL into my Ansible playbook.

Using that, renovate would extract the required info from the next line (or the regex could be prefixed with a relative line number to the comment, package and version could be on different lines) and could create MRs based on the regex matching groups. This would also enable externalizing the variables in Ansible into a variable file and use the renovate-annotation there.

If both things are not an option, where would I find the documentation on adding my own custom managers? :-P

**Comment by HonkingGoose** [comment]
This should probably be labeled `manager:ansible`? We don't have a `manager:ansible` label, or even any `ansible` labels at all. 😄 
I'll let the maintainers decide if they want a label for ansible stuff. 😉

**Comment by HonkingGoose** [comment]
> If both things are not an option, where would I find the documentation on adding my own custom managers? :-P

@dragetd Maybe any of this could help you?

Documentation on using the Ansible manager: https://docs.renovatebot.com/modules/manager/ansible/

We also have instructions on how to extend the Ansible manager `fileMatch` array here: https://docs.renovatebot.com/modules/manager/#file-matching

**Comment by dragetd** [comment]
Thanks. But I am using the Ansible manager already. It does not update in the cases I mentioned above. That is why I created this feature request. :)

I am new to TypeScript, but found the [source](https://github.com/renovatebot/renovate/tree/master/lib/manager/ansible). Not sure if I will manage to hack my suggestion myself.

**Comment by rarkins** [comment]
If it's the same files then we'd want to use a single manager. 

But as for your annotation idea - it's already supported. Check out "regexManagers" in our docs and you can look for examples in this repo too, via our org preset.

**Comment by dragetd** [comment]
Oh wow, somehow I have overlooked the regexManager. It really seems like exactly what I want. This is awesome! Damn, I love renovate. :D Sorry for the issue noise!

**Comment by rarkins** [comment]
For the benefit of future readers, we do prefer that our named managers can extract the majority of dependencies automatically, so we'd like to do ansible playbook support the full way too even if the regex manager can usually be used as a workaround.

**Comment by dragetd** [comment]
It is a bit of a rabbit hole. Ansible has a ton of modules that allow handling different kind of package managers. I gave an example of two modules that I used.

If you think it is a good idea to support more things (right now, I think it is only for docker images), we can reopen this issue.

I can also happily compile a list of what I believe the most common modules are that would match this criteria and provide an example playbook. But as I said, my TS skills are non existent, I do not really know if I can help with anything else.
